nixos/modules/services/desktops/pipewire.nix

```
@@ -61,27 +57,26 @@ in {
     assertions = [
       {
         assertion = cfg.pulse.enable -> !config.hardware.pulseaudio.enable;
-        message = "PipeWire based PulseAudio emulation doesn't use the PulseAudio service";
+        message = "PipeWire based PulseAudio server emulation replaces PulseAudio";
       }
       {
         assertion = cfg.jack.enable -> !config.services.jack.jackd.enable;
-        message = "PIpeWire based JACK emulation doesn't use the JACK service";
+        message = "PipeWire based JACK emulation doesn't use the JACK service";
       }
     ];
 
     environment.systemPackages = [ pkgs.pipewire ]
-                                 ++ lib.optional cfg.jack.enable jack-libs
-                                 ++ lib.optional cfg.pulse.enable pulse-libs;
+                                 ++ lib.optional cfg.jack.enable jack-libs;
 
-    systemd.packages = [ pkgs.pipewire ];
+    systemd.packages = [ pkgs.pipewire ] ++ lib.optional cfg.pulse.enable [ pkgs.pipewire.pulse ];
```

```suggestion
    systemd.packages = [ pkgs.pipewire ] ++ lib.optional cfg.pulse.enable pkgs.pipewire.pulse;
```
the `++` operator expects a list item, not a list

---

Or rather, we could use `lib.optionals`

---

> the ++ operator expects a list item, not a list

Actually `++` expects a list but `optional` expects a single element and returns a list (see `lib/lists.nix`):
```
optional = cond: elem: if cond then [elem] else [];
optionals = cond: elems: if cond then elems else [];
```

Just FYI, the suggestion is still correct. Nice catch!

---

The suggested way let me format it in the exact same way as the previous line so that's nice too. Not sure why I didn't see that in the first place :sweat_smile: